### PR TITLE
Coalesce

### PR DESCRIPTION
**Pull request by lehenbauer**
* add a new "coalesce" command that will return the value of the first of the one or more named variables that exists, with a default return of "" if none of the variables exist, with a -default option to set a default return value.
* fix compiler warnings and stuff

### CODE REVIEW

**Review by resuna — APPROVED** [review]
Looks like it does what it says on the box. Ship it.

File: generic/tclXcoalesce.c [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+ *-----------------------------------------------------------------------------
+ */
+
+#include "tclExtdInt.h"
+
+

+/*-----------------------------------------------------------------------------
+ * TclX_CoalesceObjCmd --
+ *     Implements the TCL chmod command:
```
Comment (wasichu) [inline generic/tclXcoalesce.c]: Should say `Implements the Tcl coalesce command`

File: generic/tclXcoalesce.c [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+ *     coalesce var ?var...? defaultValue
```
Comment (wasichu) [inline generic/tclXcoalesce.c]: This doesn't match the description of the command in the man page.  The man page says the command is:

```
coalesce ?-default value? var ?var...?
```

File: generic/tclXcoalesce.c (hunk 1 above)
Comment (lehenbauer) [reply generic/tclXcoalesce.c]: Nice catch.  Thank you.

File: generic/tclXcoalesce.c (hunk 2 above)
Comment (lehenbauer) [reply generic/tclXcoalesce.c]: Thanks here as well.  Pull request updated.

**Pull request merged**